File: leetcode/Arrays_Hash/Intersection of Two Arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        vector<int> result;
        unordered_map<int, int> count;

        for (int &num : nums1) count[num]++;

        for (int &num : nums2){
            if (count.find(num) != count.end()){
                result.push_back(num);
                count.erase(num);
            }
        }
        return result;
    }
};

File: leetcode/Arrays_Hash/Intersection of Two Arrays.py
#solution 1
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        nums1 = (set(nums1))
        nums2 = (set(nums2))

        if len(nums1) > len(nums2):
            res = nums1 - (nums1 - nums2) 
        else:
            res = nums2 - (nums2 - nums1)
            
        return list(res)
    
        ###Solution 2
        # count = {}
        # res = []

        # for num in nums1:
        #     count[num] = count.get(num, 0) + 1
        
        # for num in nums2:
        #     if num in count:
        #         res.append(num)
        #         del count[num]
        # return res     